pkg/components/data_node_remote.go

```diff
@@ -0,0 +1,93 @@
+package components
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
+
+	ytv1 "github.com/ytsaurus/ytsaurus-k8s-operator/api/v1"
+	"github.com/ytsaurus/ytsaurus-k8s-operator/pkg/apiproxy"
+	"github.com/ytsaurus/ytsaurus-k8s-operator/pkg/consts"
+	"github.com/ytsaurus/ytsaurus-k8s-operator/pkg/labeller"
+	"github.com/ytsaurus/ytsaurus-k8s-operator/pkg/resources"
+	"github.com/ytsaurus/ytsaurus-k8s-operator/pkg/ytconfig"
+)
+
+type RemoteDataNode struct {
+	server server
+	cfgen  *ytconfig.NodeGenerator
+	spec   *ytv1.DataNodesSpec
+	baseComponent
+}
+
+func NewRemoteDataNodes(
+	cfgen *ytconfig.NodeGenerator,
+	nodes *ytv1.RemoteDataNodes,
+	proxy apiproxy.APIProxy,
+	spec ytv1.DataNodesSpec,
+	commonSpec ytv1.CommonSpec,
+) *RemoteDataNode {
+	l := labeller.Labeller{
+		ObjectMeta:     &nodes.ObjectMeta,
+		APIProxy:       proxy,
+		ComponentLabel: cfgen.FormatComponentStringWithDefault(consts.YTComponentLabelDataNode, spec.Name),
+		ComponentName:  cfgen.FormatComponentStringWithDefault(string(consts.DataNodeType), spec.Name),
+	}
```

Comment:
I'm afraid you have to rebase on main since @achulkov2 refactored Labeller in his PR